mlflow/utils/uri.py

```diff
@@ -162,12 +170,62 @@ def extract_db_type_from_uri(db_uri):
     return db_type
 
 
+def _is_local_windows_path(uri_or_path):
+    """
+    Detects if platform system is windows and if `uri_or_path` is a windows path
+    """
+    parsed_uri = urllib.parse.urlparse(uri_or_path)
+    scheme = parsed_uri.scheme
+    return (
+        platform.system().lower() == "windows"
+        and not parsed_uri.netloc
+        and len(uri_or_path) > 3
+        and uri_or_path[0] in string.ascii_letters
+        and uri_or_path[0].lower() == scheme.lower()
+        and uri_or_path[1:3] in [":", ":/", "://", ":\\"]
+    )
+
+
+def _strip_leading_slash_if_windows_path(local_path):
+    """
+    Removes leading slash, if platform system is windows and `local_path` is a windows path
+
+    # For Windows path, urllib parses it with an extra leading slash
+    >>> path_with_leading_slash = urllib.parse.urlsplit("file:///C:/a/b/c").path
+    >>> path_without_leading_slash = _strip_leading_slash_if_windows_path(path_with_leading_slash)
+    >>> assert path_without_leading_slash == "C:/a/b/c"
+    """
+
+    local_parsed_path = pathlib.Path(local_path).as_posix()
+    parsed_uri = urllib.parse.urlparse(local_parsed_path)
+    if (
+        platform.system().lower() == "windows"
+        and not parsed_uri.netloc
+        and len(local_parsed_path) > 3
+        and local_parsed_path[0] == "/"
+        and local_parsed_path[1] in string.ascii_letters
+        and local_parsed_path[2:4] in [":", ":/", "://"]
+    ):
+        final_path = local_parsed_path[1:]
+        if _is_local_windows_path(final_path):
+            return final_path
```

Comment:
I'm concerned that this logic might be fairly brittle. I know we're trying to convert relative URI paths to absolute URIs paths, but I think it may be better to just do this for UNIX paths and `file://` URIs. How does that sound?